cnf-certification-test/suite_test.go

```diff
@@ -264,3 +353,209 @@ func incorporateVersions(claimData *claim.Claim) {
+	if err != nil {
+		log.Fatalf("Error writing YAML file: %v", err)
+	}
+	*/
+	t := testing.T{}
+	logrus.Infof(os.Getenv("KUBECONFIG"))
+	// When running unit tests, skip the suite
+	if os.Getenv("UNIT_TEST") != "" {
+		t.Skip("Skipping test suite when running unit tests")
+	}
+
+	err = configuration.LoadEnvironmentVariables()
+	if err != nil {
+		log.Fatalf("could not load the environment variables, error: %v", err)
+	}
+
+	// Set up logging params for logrus
+	loghelper.SetLogFormat()
+	setLogLevel()
+
+	ginkgoConfig, _ := ginkgo.GinkgoConfiguration()
+	log.Infof("Ginkgo Version      : %v", ginkgo.GINKGO_VERSION)
+	log.Infof("Labels filter       : %v", ginkgoConfig.LabelFilter)
+	// Set clientsholder singleton with the filenames from the env vars.
+	_ = clientsholder.GetClientsHolder(getK8sClientsConfigFileNames()...)
+
+	// Initialize the claim with the start time, tnf version, etc.
+	claimRoot := claimhelper.CreateClaimRoot()
+	claimData := claimRoot.Claim
+	claimData.Configurations = make(map[string]interface{})
+	claimData.Nodes = make(map[string]interface{})
+	incorporateVersions(claimData)
+
+	configurations, err := claimhelper.MarshalConfigurations()
+	if err != nil {
+		log.Errorf("Configuration node missing because of: %s", err)
+		t.FailNow()
+	}
+	claimData.Nodes = claimhelper.GenerateNodes()
+	claimhelper.UnmarshalConfigurations(configurations, claimData.Configurations)
+
+	// initialize abort flag
+	testhelper.AbortTrigger = ""
+
+	fmt.Println("This will be written to the log file.")
+	var env provider.TestEnvironment
+	env.SetNeedsRefresh()
+	env = provider.GetTestEnvironment()
+	// fetch the current config
+	suiteConfig, reporterConfig := ginkgo.GinkgoConfiguration()
+	// adjust it
+	suiteConfig.SkipStrings = []string{"NEVER-RUN"}
+	reporterConfig.FullTrace = true
+	reporterConfig.JUnitReport = "cnf-certification-tests_junit.xml"
+	// pass it in to RunSpecs
+	suiteConfig.LabelFilter = requestData.SelectedOptions
+	ginkgo.RunSpecs(&t, CnfCertificationTestSuiteName, suiteConfig, reporterConfig)
+
+	continueRun(&t, false, env, claimData, claimRoot)
+	// Return the result as JSON
+	response := struct {
+		Message string `json:"Message"`
+	}{
+		Message: fmt.Sprintf("Sucsses to run %s", requestData.SelectedOptions),
+	}
+	// Serialize the response data to JSON
+	jsonResponse, err := json.Marshal(response)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// Set the Content-Type header to specify that the response is JSON
+	w.Header().Set("Content-Type", "application/json")
+
+	// Write the JSON response to the client
+	_, err = w.Write(jsonResponse)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+}
+
+func logStreamHandler(w http.ResponseWriter, r *http.Request) {
+	conn, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		log.Printf("WebSocket upgrade error: %v", err)
+		return
+	}
+	defer conn.Close()
+
+	filePath := "log.log"
```

Comment:
Use string buffer here instead of file, remove every delays:
	for {


		scanner := bufio.NewScanner(buf)
		for scanner.Scan() {
			line := scanner.Text() + "\n"
			// Send each log line to the client
			if err := conn.WriteMessage(websocket.TextMessage, []byte(line)); err != nil {
				fmt.Println(err)
				//return
			}

			// Sleep for a short duration to simulate real-time updates

		}
		if err := scanner.Err(); err != nil {
			log.Printf("Error reading log file: %v", err)
		}

	}